From 0f50a0ef77b6f34832b6159e5afc79361cd34c77 Mon Sep 17 00:00:00 2001
From: Dien Pham <dien.pham.ry@renesas.com>
Date: Tue, 7 Feb 2017 23:24:54 +0700
Subject: [PATCH 577/909] arm64: dts: r8a7796: Add AVS device node

commit 955a72d77b7554a03b6393158b7c17ebb630b85d from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

[dien.pham.ry: arm64: dts: r8a7796: Rename target_cpus to target_devices]
This patch supports AVS (Adaptive Voltage Switching) for R8A7796 SoC.

To avoid checking for all cpu nodes,
that is redundant (e.g checking "a53_0,...")

This patch adds target_devices property to define clearly
what devices are proceeded by driver.

Signed-off-by: Van Do <van.do.xw@rvc.renesas.com>
Signed-off-by: Dien Pham <dien.pham.ry@renesas.com>
Signed-off-by: Takeshi Kihara <takeshi.kihara.df@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/renesas/r8a7796.dtsi | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm64/boot/dts/renesas/r8a7796.dtsi b/arch/arm64/boot/dts/renesas/r8a7796.dtsi
index c50d134b52cf..93e039ee1b44 100644
--- a/arch/arm64/boot/dts/renesas/r8a7796.dtsi
+++ b/arch/arm64/boot/dts/renesas/r8a7796.dtsi
@@ -786,6 +786,12 @@
 			reg = <0 0xe6060000 0 0x50c>;
 		};
 
+		avs: avs@e60a013c {
+			compatible = "renesas,r8a7796-avs", "renesas,rcar-gen3-avs";
+			reg = <0 0xe60a013c 0 0x04>;
+			target_devices = <&a57_0>, <&a57_1>;
+		};
+
 		cpg: clock-controller@e6150000 {
 			compatible = "renesas,r8a7796-cpg-mssr";
 			reg = <0 0xe6150000 0 0x1000>;
-- 
2.17.1

